### PR TITLE
Kate / DTRA-321 / TS migration of trade-params and trade-params-mobile

### PR DESCRIPTION
**Pull request by kate-deriv**
## Changes:

- TS migration of trade-params.jsx and trade-params-mobile.jsx

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx [hunk 1]
```diff
@@ -20,20 +74,21 @@ const DEFAULT_DURATION = Object.freeze({
...
+    trade_duration_unit === duration_unit
+        ? trade_duration
+        : DEFAULT_DURATION[duration_unit as keyof typeof DEFAULT_DURATION];
 
-const TradeParamsModal = observer(({ is_open, toggleModal }) => {
-    const { client, ui } = useStore();
+const TradeParamsModal = observer(({ is_open, toggleModal }: TTradeParamsModal) => {
+    const { client } = useStore();
     const { currency } = client;
-    const { enableApp, disableApp } = ui;
```
Comment (kate-deriv) [inline packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx:36]: `enableApp` and `disableApp` are not been used in `<Modal />` component, it's not expecting them. On localhost was working fine without them, so I decided to delete them. 
Feel scary.

File: packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx [hunk 2]
```diff
@@ -230,9 +283,10 @@ const TradeParamsMobile = observer(
                 onTabItemClick={setTradeParamTabIdx}
                 top
             >
-                {isVisible('duration') && (
+                {isVisible('duration') ? (
```
Comment (kate-deriv) [inline packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx:282]: `<Tab />` component were not expecting `boolean` in the children only `ReactElement` or `null`.

**Review by maryia-deriv — APPROVED** [review]

File: packages/components/src/components/tabs/tabs.tsx [hunk 3]
```diff
@@ -179,12 +179,12 @@ const Tabs = ({
...
                                 <Tab
-                                    active_icon_color={active_icon_color}
+                                    active_icon_color={active_icon_color ?? ''}
                                     className={className}
                                     count={count}
                                     icon={icon}
-                                    icon_color={icon_color}
-                                    icon_size={icon_size}
+                                    icon_color={icon_color ?? ''}
+                                    icon_size={icon_size ?? 0}
```
Comment (yashim-deriv) [inline packages/components/src/components/tabs/tabs.tsx]: You should set the fallback as a default value in the args of the component. Easier to track and more readable that way

File: packages/components/src/components/tabs/tabs.tsx [hunk 4]
```diff
@@ -14,20 +14,20 @@ declare module 'react' {
...
+    fit_content?: boolean;
     has_active_line?: boolean;
     has_bottom_line?: boolean;
     header_fit_content?: boolean;
-    history: History;
-    icon_color: string;
-    icon_size: number;
+    history?: History;
+    icon_color?: string;
+    icon_size?: number;
```
Comment (yashim-deriv) [inline packages/components/src/components/tabs/tabs.tsx]: Ensure that declaring everything as optional should have a fallback be it a condition or a default value

File: packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx [hunk 5]
```diff
@@ -12,6 +10,62 @@ import React from 'react';
...
+    currency: string;
+    toggleModal: () => void;
+    isVisible: (component_key: string) => boolean;
+    setAmountTabIdx: (amount_tab_idx?: number) => void;
+    amount_tab_idx?: number;
+    setTradeParamTabIdx: (trade_param_tab_idx: number) => void;
+    trade_param_tab_idx: number;
+    setDurationTabIdx: (duration_tab_idx?: number) => void;
+    duration_unit: string;
+    duration_units_list: Array<TTextValueStrings>;
```
Comment (yashim-deriv) [inline packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx]:
```suggestion
    duration_units_list: TTextValueStrings[];
```
nit: looks more readable this way?

File: packages/trader/src/Modules/Trading/Containers/trade-params.tsx [hunk 6]
```diff
@@ -17,14 +16,21 @@ import { observer } from '@deriv/stores';
...
     const { form_components } = useTraderStore();
-    const isVisible = component_key => {
+    const isVisible = (component_key: string) => {
         return form_components.includes(component_key);
     };
     return (
         <React.Fragment>
-            {isVisible('duration') && <Duration key={'duration'} is_minimized={is_minimized} />}
+            {isVisible('duration') && (
+                // @ts-expect-error: we expect error here because <Duration /> is not in TS yet
```
Comment (yashim-deriv) [inline packages/trader/src/Modules/Trading/Containers/trade-params.tsx]: Mark all ts ignore with TODO. It is easier to track for refactoring in the future

**Review by yashim-deriv — CHANGES_REQUESTED** [review]

**Review by balakrishna-deriv — APPROVED** [review]

File: packages/components/src/components/tabs/tabs.tsx (hunk 3 above)
Comment (kate-deriv) [reply packages/components/src/components/tabs/tabs.tsx]: Will do, thank you

File: packages/components/src/components/tabs/tabs.tsx (hunk 4 above)
Comment (kate-deriv) [reply packages/components/src/components/tabs/tabs.tsx]: I'll add default value

**Review by yashim-deriv — APPROVED** [review]

**Review by maryia-deriv — APPROVED** [review]

File: packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx [hunk 7]
```diff
@@ -12,6 +10,62 @@ import React from 'react';
 import classNames from 'classnames';
 import { localize } from '@deriv/translations';
 
+type TTradeParamsModal = {
+    is_open: boolean;
+    toggleModal: () => void;
+};
+
+type TTextValueStrings = {
```
Comment (henry-deriv) [inline packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx]: i'll put this type into `common-prop.type.ts` and export. Just pull my branch and import.

**Review by henry-deriv — APPROVED** [review]

**Pull request merged**